tests/test_logging.py

```diff
@@ -817,14 +660,75 @@ def assert_stats_logs(
     log_output,
     expected_initial_study_def_logs,
     expected_timing_log_params,
+    downloaded=True,
 ):
-    all_stats_logs = get_stats_logs(log_output.entries)
+    # get the cohortextractor-stats logs
+    cohortextractor_stats_logs = get_stats_logs(log_output.entries)
     for log_params in expected_initial_study_def_logs:
-        assert log_params in all_stats_logs
-
-    timing_logs = get_logs_by_key(log_output.entries, "timing_id")
+        assert log_params in cohortextractor_stats_logs
 
+    timing_logs = get_logs_by_key(cohortextractor_stats_logs, "timing_id")
     assert len(timing_logs) == len(expected_timing_log_params), timing_logs
+
+    # get the sqlserver-stats logs
+    sqlserver_stats_logs = get_stats_logs(log_output.entries, event="sqlserver-stats")
+
+    # Each SQL execute call has a timing id, and should be associated with
+    # one SQL Server parse/compile log and one execution log;
+    # **EXCEPT** CREATE INDEX and DROP TABLE queries, which don't create a parse/compile
+    # timing, only an execution timing
+
+    # However, when we generate the queries to select the final results and write
+    # them to file, we return a generator.  At the point where the query is actually
+    # executed and the timing message handler is called, the last timing id is the one for
+    # dropping the temporary results table query. So we get no execution_time log for the
+    # download queries' timing_ids, and the DROP TABLE query's timing id will collect execution timing for
+    # both the download queries and (finally) the real drop table query
```

Comment:
arg, this is a bit of a mess isn't it.

I suspect there's a better way to do this tracking, but it escapes me on a Friday afternoon.

Reply:
I'm sure there must be too, but I couldn't find one